### ISSUE TITLE
pip install fails: incorrect metadata

### ISSUE THREAD

**Issue by chstem** [deepchem/deepchem#2310]
Installing deepchem using pip 20.3 results in the following error message:

```
$ pip install deepchem 
Collecting deepchem
  Using cached deepchem-2.4.0rc1.dev20201119053907.tar.gz (401 kB)
ERROR: Requested deepchem from https://files.pythonhosted.org/packages/fb/32/087
7d24af313f071d7b2b93765c5759a861b46e603176d21bb339e2013cd/deepchem-2.4.0rc1.dev2
0201119053907.tar.gz#sha256=e886573f12a52fd62a460835629daa4f2236e0a2926ec9766b7f
02c89fc8412b has different version in metadata: '2.4.0rc1.dev20201204104606'  
```

Apparently, there are two slightly different timestamps used in the packaging of a deepchem release for pypi.

This seems to be related to the new resolver used by pip, since using pip version 20.2 this error only occurs if `--use-feature=2020-resolver` is used.

**Comment by rbharath** [comment]
Hmm, this is unfortunate. It seems we might need to tweak our pip infrastructure to be more robust to the new resolver.

@nd-02110114 Would you have any thoughts off the top of your head? I think you've been looking at upgrading the pip infrastructure recently

**Comment by LarsAC** [comment]
For use of pip 20.3 I could solve using additional option `--use-deprecated=legacy-resolver`. As per pip documentation, this option will go away in 21.0, though.

**Comment by nd-02110114** [comment]
As @LarsAC mentioned, the workaround is the following command.

```
$ pip install --use-deprecated=legacy-resolver --pre deepchem
```

The reason why this error happen is that we publish only a tar.gz file through PyPI, not a wheel file.

I understood how `pip install` works from this URL : https://pydist.com/blog/pip-install.
When we run `pip install some_package`, firstly the PyPI resolver searches a `wheel file` which is a built-package format. If the resolver don't find a `wheel file`, it searches a `tar.gz file` and build some_package.  In the case of `tar.gz file`, the resolver need to build the package. (On the other hand, there is no need in the case of `wheel`). 

Now, DeepChem published only `tar.gz file`, so PyPI resolver find `deepchem-2.4.0rc1.dev20201119053907.tar.gz`(latest package) and build it when running `pip instal --pre deepchem`. But, the package version which is built by PyPI will be `deepchem-2.4.0rc1.dev.build_time` (built_time is the time when installing) based on out versioning logic in setup.py. So, PyPI raise the reported error (version mismatch).

I think this issue will be resolved by publishing `wheel file`. The `wheel file` is more appropriate file format when publishing the package. This is because the `wheel file` is faster to install without building.

I will try to fix in #2290

**Comment by nd-02110114** [comment]
The installation from `tar.gz file`

<img width="1269" alt="スクリーンショット 2020-12-10 15 16 26" src="https://user-images.githubusercontent.com/17228098/101793730-749c8c00-3b49-11eb-80ec-bf1d947c97aa.png">



The installation from `wheel file`

<img width="1273" alt="スクリーンショット 2020-12-10 15 29 15" src="https://user-images.githubusercontent.com/17228098/101793717-723a3200-3b49-11eb-9c11-5f4171506979.png">

**Comment by nd-02110114** [comment]
I seem this issue was resolved by #2335
